### ISSUE TITLE
Possible issue with Scratch API/HTTPS

### ISSUE THREAD

**Issue by oskay** [evil-mad/robopaint#299]
TLDR: Control from Snap (and possible other similar interfaces) is currently broken.

Filing this under Robopaint for the moment, but may be more appropriate under https://github.com/techninja/cncserver/issues/ .

With RP open, if I run commands like `localhost:4242/pen.up` in a browser, that works well. However, trying out the same command in [Snap](http://snap.berkeley.edu), as with the https://github.com/evil-mad/WaterColorBlocks examples, does not work with current versions of Snap.

On older versions of Snap (pre-4.1), there was a `http://` block, to retrieve the specified URL (less protocol), that worked quite well.

With Snap 4.1, the block was changed to `url`, and retrieves the specified URL, but defaults to https if no protocol is given. You can get around this by explicitly calling the protocol, as in: `http://localhost:4242/poll`-- and we've just rolled that set of changes to [WaterColorBlocks](https://github.com/evil-mad/WaterColorBlocks). 

Unfortunately, that's only _part_ of the solution. Adding the explicit `http://` allows Snap to work if (and only if) it's downloaded to your computer. (Download a source code zip from https://github.com/jmoenig/Snap--Build-Your-Own-Blocks/releases and open snap.html .) 

Normally, running snap from within your browser, one runs it online, and under https. My impression is that this will not allow an http connection to localhost. (See here, for example: https://github.com/jmoenig/Snap--Build-Your-Own-Blocks/issues/2048#issuecomment-370790060 )
